source/extensions/filters/http/cache/simple_http_cache/simple_http_cache.cc

```diff
@@ -183,7 +183,7 @@ void SimpleHttpCache::varyInsert(const Key& request_key,
   if (iter == map_.end()) {
     Http::ResponseHeaderMapPtr vary_only_map =
         Http::createHeaderMap<Http::ResponseHeaderMapImpl>({});
-    vary_only_map->setCopy(Http::Headers::get().Vary, vary_header->value().getStringView());
+    vary_only_map->setCopy(Http::Headers::get().Vary, vary_header[0]->value().getStringView());
```

Comment:
shouldn't this support multi-value headers?
Maybe a TODO for caching folks to look into this one?